app/controller.py

```diff
@@ -22,17 +25,46 @@ def scan_image():
 
     file = request.files['file']
 
+    if file.filename == '':
+        return jsonify({"error": "No selected file"}), 400
+
+    if not allowed_file(file.filename):
+        return jsonify({"error": "File type not allowed. Only images are allowed."}), 400
+
+    vision_service = VisionService()
+    image_content = file.read()
+
+    try:
+        detected_text = vision_service.detect_text_in_image(image_content)
+        repository.save_receipt(receipt_content=detected_text)
+        return jsonify({'detected_text': detected_text})
+
+    except Exception as e:
+        print(f"Error occurred: {e}")
+        return jsonify({'error': str(e)}), 500
+
+@deprecated("This function probably will be removed in future versions")
+@doc_scanner_bp.route('/scan2', methods=['POST'])
+def scan_image2():
+    if 'file' not in request.files:
+        return jsonify({"error": "No file part"}), 400
+
+
+    file = request.files['file']
+
     if file.filename == '':
         return jsonify({"error": "No selected file"}), 400
 
     if not allowed_file(file.filename):
         return jsonify({"error": "File type not allowed. Only images are allowed."}), 400
 
     image = Image.open(file.stream)
+
     image_np = np.array(image)
 
     scanned_image = run_scan_by_image(image_np)
 
+
     # TODO -> convert to text then save it in db
```

Comment:
co w końcu z tym TODO?

Reply:
To w deprecated. Więc cała metoda możliwe, że do wywalenia chyba, że inna decyzja będzie

Reply:
A spróbuj tam zrobić tak, żeby faktycznie zapisał się tekst do bazy przy tym endpoincie. Tylko do bazy, nie trzeba go zwracać

Reply:
Zapisuje się przecież, tylko w tym endpoincie wyzej Ten o ktorym piszemy tutaj to do usuniecia jest ' -- '

Reply:
Moim zdaniem, do przerobienia, że na razie zwracamy zdjęcie a text zapisujemy do bazy danych. Taki był zamysł z tego co kojarze

Reply:
Zrobione